src/server/bg_services/log_replication_scanner.js

```diff
@@ -188,6 +193,17 @@ class LogReplicationScanner {
         }
     }
 
+    async copy_object(src_bucket_name, dst_bucket_name, key) {
+        await replication_utils.copy_objects(
+            this._scanner_sem,
+            this.client,
+            src_bucket_name.unwrap(),
+            dst_bucket_name.unwrap(),
+            [key],
```

Comment:
it is not something new in this PR, but I think we should improve the way we process candidates. 
instead of processing it one by one (starting [here](https://github.com/achouhan09/noobaa-core/blob/98347453f3cbf34cbde42b2246a5c02dd56398f8/src/server/bg_services/log_replication_scanner.js#L133) ), we should create a batch of copy and delete keys and send it to the replication server in one call.

@achouhan09 @utkarsh-pro WDYT?

Reply:
Sure, will work on it.

Reply:
As per discussion, will create a separate PR for this.